File: baekjoon/7568/7568.py
'''
덩치 python 풀이
https://doctcoder.tistory.com/entry/%EB%8D%A9%EC%B9%98-%ED%92%80%EC%9D%B4
'''
n=int(input())
x=[]
y=[]
rec=[1 for i in range(n)]
for i in range(n):
	a,b=map(int,input().split())
	x.append(a)
	y.append(b)
for i in range(n):
	for j in range(n):
		if x[i]>x[j] and y[i]>y[j]:
			rec[j]+=1
for i in range(n):
	print(rec[i],end=" ")

File: baekjoon/7568/7568.cpp
/*
덩치 C++ 풀이
https://doctcoder.tistory.com/entry/%EB%8D%A9%EC%B9%98-%ED%92%80%EC%9D%B4
*/
#include <iostream>
#include <vector>
using namespace std;
int main()
{
  int n,a,b;
  cin>>n;
	vector<int> weight;
	vector<int> height;
	vector<int> ranks(n);
	fill(ranks.begin(),ranks.end(),1);
	for(int i=0;i<n;i++)
	{
		cin>>a>>b;
		weight.push_back(a);
		height.push_back(b);
	}
	for(int i=0;i<n;i++)
	{
		for(int j=0;j<n;j++)
		{
			if(weight[i]>weight[j]&&height[i]>height[j]) ranks[j]++;
		}
	}
	for(int i=0;i<n;i++)
	{
		cout<<ranks[i]<<" ";
	}
}
